doc/nyan/base types.md

```diff
@@ -0,0 +1,164 @@
+	# And this would allow to reference
+
+	# WIP4: how do we reference Animations?
+#	animation : file # one picture that contains all frames or, if a directory,
+	                # contains all frames. # WIP5: naming scheme?
+#	renameMe : file # WIP6: how to read `animation'? describes, where in the big picture `animation' which frames are; whether to potentially mirror them.
+	                # possibilities: 1. $renameMe is a regular file: description of `animation' (syntax?)
+	                # 2. replace `renameMe' by more fields that describe `animation' (ugly imo, but maybe the best way)
+	                # 3. remove `renameMe':
+	                #    if $animation is a directory: look for '$animation/animate.nfo'
+	                #    else: look for "$(basename $animation png)nfo"
+
+CarryAbility(Ability): # WIP: find a better name?
+	# Used by priests for relicts.
+	# WIP1: how should this look like? keep it general? can maybe be used by buildings to let units find shelter in them? or by mods for adding new relict-like objects
+	# WIP2: can maybe merged with a modified GatherAbility?
+
+#	goodMemberName : Portable # what can be collected? make a 'Portable' class and let relict inherit that? could then be also used by resources?
+#	maxAmount
+
+	animation : file # what does a priest look like when he's carrying it?
+	                 # WIP: see GatherAbility.animation and nyan question 4 (further down).
+
+AttackAbility(Ability):
+	# Every attack shall inherit from this.
+	# WIP: how will damage calculation work in oa?
+
+	range : float
+	rate : float
+	damage : float # rename to baseDamage ?
+
+
+Unit(): # WIP
+	# Everything you can see in the game # (WIP: but not assets, which is basically everything you see. Maybe better:)
+	# Everything that has some 'Ability' inherits from this.
+
+	ability : set(Ability)
+
+
+
+#and probably more. if you know what's missing, add it.
+
+```
+
+Open nyan implementation questions:
+
+1. How do we say in Nyan, that a Towncenter attack depends on how many people are inside the building?
+2. What about the building 'Ability' to carry units? will this lead to recursive references?
+3. How do we reference hooks from Nyan/create new hooks in mods?
+
+  -> If the {Towncenter,Castle} has certain damage, contained units need to leave.
+  
+  -> A relict in a Church leads to increasing gold.
+4. How do we add new animations?
+
+  -> How would a mod add a new relict-like object that can be carried (the animation)? can we describe composition of frames? How is it done by the original game?
+  
+  -> We need a unified way/member-name to add a new animation to an 'Ability'
+  
+  -> animations are stored in one big picture (or allow one file per frame?). How do we tell where in the picture is which frame? How can you tell to mirror a frame?
+5. Of what type are the ingame objects, that don't move and give ressources? goldmine, fish, dead deer? How are deer and dead deer connected?
```

Comment:
yes, could also work. I think you're right that all units have a dying animation but many won't have one. also we somehow need to model that the living deer is not a resource spot.

Basically we have multiple possibilities to model this:
* store everything relevant for the dead deer in the living one already, aka `rotting_animation` and `resource_decay` and `resource_type` etcetc, this is the "bad" approach i'd say
* have a separate `DeadDeer` that inherits from `(original deer, resource spot)` and updates graphics etc. the original deer then references to that object as `on_death_become : Unit = DeadDeer`. this requires a forward declaration which will be allowed in the next revision of the nyan design.
* use abilities to test if the deer can even die. that ability then performs the death animation, another ability performs the resource collection, another one the decay.

I think the best approach is having an ability "being hunted" because it's different if a villager or a tribok killed it for the collectible food. That ability is implemented to change the unit type to `DeadDeer` once it is killed. The ability is included in the `Deer` `abilities` set, and also requires a forward declaration. It's the most overkill but also most extensible and awesome approach i'd say.